src/components/Wallet.jsx

```diff
@@ -100,47 +133,89 @@ export default function Wallet({ name, currentWallet, startWallet, stopWallet, s
       const { action, password } = serialize(form)
 
       switch (action) {
-        case 'unlock':
-          unlockWallet(name, password)
+        case 'unlock': {
+          if (currentWallet) {
+            setAlert({
+              variant: 'warning',
+              message:
+                currentWallet.name === name
+                  ? // unlocking same wallet
+                    t('wallets.wallet_preview.alert_wallet_already_unlocked', { walletName: walletDisplayName(name) })
+                  : // unlocking another wallet while one is already unlocked
+                    t('wallets.wallet_preview.alert_other_wallet_unlocked', { walletName: walletDisplayName(name) }),
+            })
+          } else {
+            unlockWallet(name, password)
+          }
           break
-        case 'lock':
-          lockWallet()
+        }
+        case 'lock': {
+          const needsLockConfirmation = coinjoinInProgress || makerRunning
```

Comment:
Hmm.. works on my end. Of course, it might be that the state is not updated, e.g. when the websocket is not connected (and the next poll request to `/session` has not finished yet). So the session data must not be considered as absolute truth, but just tested the "happy path" and it worked as expected.. :thinking:

Reply:
I see, I was probably to fast. Waiting a couple of seconds after activating _earn_ works. ✅

Reply:
This is not optimal either. Feels like it waited for the `/session` poll request to finish, whereas the websocket update should provide the `maker_running` information "immediately" - at earlier than the polling request most of the time. This is what
```ts
type SessionInfoUpdate = SessionInfo | MakerRunningFlag | CoinjoinInProgressFlag
```
should emphasize: Either update the whole `SessionInfo` or some parts of it(`MakerRunningFlag | CoinjoinInProgressFlag`) -> see [updating the info on websocket data in SessionInfoContext.tsx#L101-L112](https://github.com/joinmarket-webui/joinmarket-webui/blob/lock-unlock-handling2/src/context/SessionInfoContext.tsx#L101-L112)